client/app/main/student/student.controller.js

```diff
@@ -2,9 +2,30 @@
 
 var app = angular.module('app');
 
-function StudentCtrl($scope, $state, $stateParams, Student, toastr) {
+function StudentCtrl($scope, $state, $stateParams, Student, 
+  AbsenceRecord, toastr) {
   $scope.student = Student.get({id: $stateParams.id});
 
+  $scope.calcPercentage = function() {
+    AbsenceRecord.current()
+      .$promise.then(function(records) {
+        var totPresent = [];
+        var totEnrolled = [];
+        var addToNext = function(prev, cur) {
+          return prev + cur;
+        };
+        records.forEach(function(record) {
+          totPresent.push(record.entries.present);
+          totEnrolled.push(record.entries.enrolled);
+        });
+        $scope.percentage = Math.floor(
+          (totPresent.reduce(addToNext) / totEnrolled.reduce(addToNext))
+          * 100);
+      });
+  };
+
+  $scope.calcPercentage();
+
```

Comment:
Is there a function in the AbsenceRecord service or somewhere else that aggregates all the data for a single student? I couldn't find it. I guess I made the wrong assumption that  `AbsenceRecord.current()` would provide the data for the selected student.